tests/validator/validator.cpp

```diff
@@ -698,70 +1448,349 @@ TEST(Validator, parseAndValidateInvalidUnitErrors) {
+    m->addComponent(comp3);
+    m->addComponent(comp4);
+    m->addComponent(comp5); 
+    m->addComponent(comp6);
+    m->addComponent(comp7);
+    m->addComponent(comp8);
+    m->addComponent(comp9); 
+    m->addComponent(comp10);
+    m->addComponent(comp11);
+    m->addComponent(comp12);
+    m->addComponent(comp13);
+    m->addComponent(comp14);
+    m->addComponent(comp15);
+    m->addComponent(comp16);
+    m->addComponent(comp17);
+
+    libcellml::Variable::addEquivalence(v2, v4); 
+    libcellml::Variable::addEquivalence(v3, v4);
+    libcellml::Variable::addEquivalence(v8, v6); 
+    libcellml::Variable::addEquivalence(v2, v8);    
+    libcellml::Variable::addEquivalence(v3, v6);    // Loop formed: 2, 4, 3, 6, 8, 2 
+
+    libcellml::Variable::addEquivalence(v6, v7);    
+    
+    // NB Attached loops (ie: a-b-c-a and b-c-d-b) *are* detected and handled but return too many variations of the loop 
+    // (eg a-b-c-d, a-b-c, b-c-d, etc) to be viable as a hard-coded test, so have broken the attachment here.
+    // libcellml::Variable::addEquivalence(v3, v7);    // Second attached loop: 3, 6, 7 
+
+    libcellml::Variable::addEquivalence(v9, v5); 
+    libcellml::Variable::addEquivalence(v5, v1_2);
+    libcellml::Variable::addEquivalence(v1_2, v9);  // Unattached loop: 5, 9, 1_2
+
+    libcellml::Variable::addEquivalence(v11, v4); 
+    libcellml::Variable::addEquivalence(v11, v14);
+    libcellml::Variable::addEquivalence(v11, v13);
+    libcellml::Variable::addEquivalence(v11, v12);
+    libcellml::Variable::addEquivalence(v12, v15);
+    libcellml::Variable::addEquivalence(v12, v17);
+    libcellml::Variable::addEquivalence(v17, v16);   // Complicated tree, no loops 
+
+    libcellml::Variable::addEquivalence(v1_1, v8);  
+    libcellml::Variable::addEquivalence(v1_3, v2);  
+    // libcellml::Variable::addEquivalence(v1_4, v2);  // TODO Can two sibling variables in the same component be equivalent to one in another?
+
+    v.validateModel(m);
+    EXPECT_EQ(1u, v.errorCount());
+
+    // Two loops are present, split error message at "Loop":
+    // Check that one loop contains 'variable6' <-> 'variable8' <-> 'variable2' <-> 'variable4' <-> 'variable3' <-> 'variable6',
+    // Check that the other contains 'variable1_2' <-> 'variable9' <-> 'variable5' <-> 'variable1_2',
+    // "Cyclic variables exist, 2 loops found (Variable(Component)). Loop: 'variable4'(in 'component4') <-> 'variable3'(in 'component3') <-> 'variable6'(in 'component6') <-> 'variable8'(in 'component8') <-> '...
+    if (v.errorCount() == 1) {
+        size_t pos = 0;
+        std::string split = "Loop: ";
+        std::string msg = v.getError(0)->getDescription();
+        std::vector<std::string> split_msg;
+
+        while ((pos = msg.find(split)) != std::string::npos) {
+            split_msg.push_back(msg.substr(0, pos));
+            msg.erase(0, pos + split.length());
+        }
+        split_msg.push_back(msg);
+
+        EXPECT_EQ(split_msg.size(), 3);
```

Comment:
expected answer first, actual answer second.  3u anyway.

Reply:
`3u` for now, but it will eventually become `size_t(3)` whenever PR #288 gets merged in.

Reply:
I'll fix this to be consistent with 3u now, we can do a mass substituion for the size_t(3) syntax separately.